test/run_test.py

```diff
@@ -920,33 +928,38 @@ def run_doctests(test_module, test_directory, options):
     return result
 
 
-def print_log_file(test: str, file_path: str, failed: bool) -> None:
-    num_lines = sum(1 for _ in open(file_path, "rb"))
+def sanitize_file_name(file: str):
+    return file.replace("\\", ".").replace("/", ".").replace(" ", "_")
+
+
+def handle_log_file(
+    test: ShardedTest, file_path: str, failed: bool, was_rerun: bool
+) -> None:
     test = str(test)
-    n = 100
     with open(file_path, "rb") as f:
+        full_text = f.read().decode("utf-8", errors="ignore")
+
+    if not failed and not was_rerun and "=== RERUNS ===" not in full_text:
```

Comment:
Do we do reruns for C++ tests as well?

Reply:
I don't think so but cc @huydhn to confirm

Reply:
We actually have stepcurrent retry for C++ test.  That plugin works with pytest-cpp last time I check.  I don't have an actual example of C++ test retrying at hand (they are pretty reliable), but the successful C++ tests output is truncated in the same way here from what I see.